rust/main/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -663,14 +663,19 @@ impl HyperlaneDb for HyperlaneRocksDB {
         &self,
         message_id: &H256,
         payload_id: &UniqueIdentifier,
-    ) -> DbResult<()> {
-        self.store_value_by_key(PAYLOAD_ID_BY_MESSAGE_ID, message_id, payload_id)
+    ) -> DbResult<Vec<UniqueIdentifier>> {
+        let mut ids = self
+            .retrieve_payload_id_by_message_id(message_id)?
+            .unwrap_or_default();
+        ids.push(payload_id.clone());
```

Comment:
I would instead rename this function to `store_payload_ids_by_message_id`, take in a vector of `UniqueIdentifier`, and write to the DB without reading first. All payloads associated with a message would be known from when the message is first processed, so there'd be no need to add extra payloads for a message later

Reply:
Let me see how [this task](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5894) should be implemented and understand if we need to change it.

Reply:
I see what you mean. We shall create several payloads for the same message and we don't need them anymore if message is returned back to prepare queue.

Reply:
Fixed